### PR TITLE
Csv import: Refactorings & CsvMapping class

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
This breaks off a few changes from #9437 for the CSV import

## How Has This Been Tested?

The main change was tested, this should purely be adding code, tested bugfixes and nonfunctional refactorings

## Learning (optional, can help others)
I'm unlikely to going to get a large change in in a single PR, and it's better to break it down, despite feeling it's readable as-is

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
lint check is possibly a remnant as you teased the commits apart?

```
Lint found 2 errors. First failure:
Fix the issues identified by lint, or add the following to your build script to proceed with errors:
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/drawable/edit_text_color_enabled_selector.xml:20: Error: The resource R.drawable.edit_text_color_enabled_selector appears to be unused [UnusedResources]
...
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckSpinnerSelection.java [hunk 1]
```diff
@@ -91,7 +92,7 @@ public void initializeActionBarDeckSpinner(@NonNull ActionBar actionBar) {
 
     }
 
-    public void initializeNoteEditorDeckSpinner(@NonNull Card currentEditedCard, boolean addNote) {
+    public void initializeNoteEditorDeckSpinner(@Nullable Card currentEditedCard, boolean addNote) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckSpinnerSelection.java:103]: verified variable is null-checked internally in method :heavy_check_mark:

**Review by mikehardy — CHANGES_REQUESTED** [review]
Read through it all - seems okay. I'm a bit unclear on major vs minor changes (dispatched where? who would use those handlers + how do you add them to use the facility? I didn't see it in the test so was just a bit unclear). One method name thought, but otherwise LGTM + want to merge

File: AnkiDroid/src/main/java/com/ichi2/utils/SequenceUtil.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.utils
+
+object SequenceUtil {
+    // https://stackoverflow.com/a/56059647/
+    /** Note: If the predicate returns false on the first element, it will be returned */
+    fun <T> Sequence<T>.takeWhileInclusive(predicate: (T) -> Boolean) = sequence {
```
[the remark this answers is not in the document]
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/utils/SequenceUtil.kt:22]: Nowhere near bikeshedding, perfect feedback, thanks!

**Comment by david-allison-1** [comment]
I'll add more documentation on the dispatch methods. I'd rather the code spoke for itself than explain it here. Thanks!

**Comment by david-allison-1** [comment]
Awaiting re-review (assuming I didn't blow up CI), especially to see if the distinction between major and minor changes has been made more clear.

If not, I'll explain the usages and let's brainstorm some better docs.

**Review by mikehardy — APPROVED** [review]
All LGTM - boom

**Comment by mikehardy** [comment]
was waiting on this one for a re-sync on I18n now I'll do an alpha with fresh strings and all the merges of the day

**Pull request merged**